docs/service-fabric/refactor-migrated-app.md

```diff
@@ -0,0 +1,160 @@
+
+## Design considerations
+ 
+The following diagram shows the architecture of the Surveys application refactored to a more granular architecture:
+
+![](./images/surveys_03.png)
+
+**Tailspin.Web** is a stateless service self-hosting an ASP.NET MVC application that Tailspin customers visit to create surveys and view survey results. This service shares most of its code with the *Tailspin.Web* service from the ported Service Fabric application. As mentioned earlier, this service uses ASP.NET core and switches from using Kestrel as web frontend to implementing a WebListener.
+
+**Tailspin.Web.Surveys.Public** is a stateless service also self-hosting an ASP.NET MVC site. Users visit this site to select surveys from a list and then fill them out. This service shares most of its code with the *Tailspin.Web.Survey.Public* service from the ported Service Fabric application. This service also uses ASP.NET Core and also switches from using Kestrel as web frontend to implementing a WebListener.
+
+**Tailspin.SurveyResponseService** is a stateful service that stores survey answers in Azure Blob Storage. It also merges answers into the survey analysis data. The service is implemented as a stateful service because it uses a [ReliableConcurrentQueue][reliable-concurrent-queue] to process survey answers in batches. This functionality was originally implemented in the *Tailspin.Web.Survey.Public* service in the ported Service Fabric application. Tailspin refactored the original functionality into this service to allow it to scale independently.
+
+**Tailspin.SurveyManagementService** is a stateless service that stores and retrieves surveys and survey questions. The service uses Azure Blob storage. This functionality was also originally implemented in the *Tailspin.AnswerAnalysisService* service in the ported Service Fabric application. Tailspin refactored the original functionality into this service to also allow it to scale independently.
+
+**Tailspin.SurveyAnswerService** is a stateless service that retrieves survey answers and survey analysis. The service also uses Azure Blob storage. This functionality was also originally implemented in the *Tailspin.AnswerAnalysisService* service in the ported Service Fabric application. Tailspin refactored the original functionality into this service because it expects less load and wants to use fewer instances to conserve resources.
+
+**Tailspin.SurveyAnalysisService** is a stateless service that persists survey answer summary data in a Redis cache for quick retrieval. This service is called by the *Tailspin.SurveyResponseService* each time a survey is answered and the new survey answer data is merged in the summary data. This service includes the functionality remaining in the *Tailspin.SurveyAnalysisService* service from the ported Service Fabric application.
```

Comment:
"This service includes the functionality remaining in the *Tailspin.SurveyAnalysisService* service from the ported Service Fabric application."

Should be...

"This service includes the functionality remaining in the *Tailspin.AnswerAnalysisService* service from the ported Service Fabric application."